src/Controller/Renaissance/Formation/ContentController.php

```diff
@@ -28,8 +29,12 @@ public function __construct(
 
     public function __invoke(Formation $formation): Response
     {
-        $formation->incrementPrintCount();
-        $this->entityManager->flush();
+        /** @var Adherent $adherent */
+        $adherent = $this->getUser();
+
+        if (!$formation->hasPrintByAdherent($adherent) && $formation->addPrintByAdherent($adherent)) {
```

Comment:
we can simplify this by removing `$formation->hasPrintByAdherent` call as we already do this check in `addPrintByAdherent` method.

```php
if ($formation->addPrintByAdherent($adherent)) {
```